synapse/storage/devices.py

```diff
@@ -634,19 +641,42 @@ def _mark_as_sent_devices_by_remote_txn(self, txn, destination, stream_id):
         txn.execute(sql, (destination, stream_id,))
 
     @defer.inlineCallbacks
-    def get_user_whose_devices_changed(self, from_key):
+    def get_user_whose_devices_changed(self, user_id, from_key):
         """Get set of users whose devices have changed since `from_key`.
         """
         from_key = int(from_key)
         changed = self._device_list_stream_cache.get_all_entities_changed(from_key)
         if changed is not None:
-            defer.returnValue(set(changed))
+            changed_set = set(changed)
+            if self._attestation_stream_cache.has_entity_changed(user_id, from_key):
+                # the user's attestations have changed, so fetch the new users
+                # that have been attested
+                sql = """
+                    SELECT DISTINCT user_id FROM attestations_stream
+                     WHERE from_user_id = ? AND stream_id > ?
+                """
+                rows = yield self._execute(
+                    "get_user_whose_devices_were_attested", None, sql, user_id, from_key
+                )
+                changed_set.update(row[0] for row in rows)
+            defer.returnValue(changed_set)
 
+        # get device updates
         sql = """
             SELECT DISTINCT user_id FROM device_lists_stream WHERE stream_id > ?
         """
         rows = yield self._execute("get_user_whose_devices_changed", None, sql, from_key)
-        defer.returnValue(set(row[0] for row in rows))
+        changed_set = set(row[0] for row in rows)
+        # get attestations that the user has made
```

Comment:
is it deliberate that this comment says something completely different to the one at line 652? if not, which one is right?